source/python/deps/deps_cli.py

```diff
@@ -915,85 +917,104 @@ def cli(
-            here=here,
-            jobs=jobs,
-            jobs_unordered=jobs_unordered,
-        )
+    with setup_deps_work_dir():
+        try:
+            if force_color:
+                _click_echo_color = True
+                if sys.platform == "win32":
+                    # Click always wrap the output stream on windows calling
+                    # `click.utils.auto_wrap_for_ansi`, setting to `None` causes ansi escape codes to
+                    # be output.
+                    click.utils.auto_wrap_for_ansi = None
+
+            directories = find_directories(project)
+
+            root_deps = obtain_all_dependecies_recursively(
+                directories, ignore_project, skip_project
+            )
+            if repos:
+                root_deps = obtain_repos(root_deps)
+
+            if pretty_print:
+                # We don't need them in order to pretty print.
+                pretty_print_dependency_tree(root_deps)
+                return 0
+
+            def required_files_filter(dependency, quiet):
+                """
+                :type dependency: Dep
+                :type quiet: bool
+
+                :return: `True` if the necessary files/folders are present, `False` otherwise.
+                """
+                for f in require_file:
+                    file_to_check = os.path.join(
+                        dependency.abspath, format_command(f, dependency)
+                    )
+                    if not os.path.isfile(file_to_check) and not os.path.isdir(
+                        file_to_check
+                    ):
+                        if not quiet:
+                            msg = '{}: skipping since "{}" does not exist'
+                            msg = msg.format(dependency.name, file_to_check)
+                            click.secho(msg, fg="cyan", color=_click_echo_color)
+                        return False
+                return True
+
+            deps_in_order = obtain_dependencies_ordered_for_execution(root_deps)
+
+            if deps_reversed:
+                deps_in_order = list(reversed(deps_in_order))
+
+            if not command:
+                deps_to_output = [
+                    dep.name
+                    for dep in deps_in_order
+                    if not dep.ignored
+                    and not dep.skipped
+                    and required_files_filter(dep, quiet=True)
+                ]
+                print("\n".join(deps_to_output))
+                return 0
+
+            # Execution.
+            execution_return = execute_command_in_dependencies(
+                command,
+                deps_in_order,
+                required_files_filter=required_files_filter,
+                dry_run=dry_run,
+                verbose=verbose,
+                continue_on_failure=continue_on_failure,
+                here=here,
+                jobs=jobs,
+                jobs_unordered=jobs_unordered,
+            )
```

Comment:
I think you could have put the context just around this function call, narrowing the context.

No command execution (where the foler could be used) happens anywhere in this scope.

Reply:
Will do.